dio/lib/src/transformers/sync_transformer.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import '../adapter.dart';
+import '../headers.dart';
+import '../options.dart';
+import '../transformer.dart';
+
+@Deprecated('Use BackgroundTransformer instead')
+typedef DefaultTransformer = SyncTransformer;
+
+/// The callback definition for decoding a JSON string.
+typedef JsonDecodeCallback = FutureOr<dynamic> Function(String);
+
+/// The callback definition for encoding a JSON object.
+typedef JsonEncodeCallback = FutureOr<String> Function(Object);
+
+/// The default [Transformer] for [Dio].
+///
+/// If you want to custom the transformation of request/response data,
+/// you can provide a [Transformer] by your self, and replace
+/// the [DefaultTransformer] by setting the [dio.transformer].
+class SyncTransformer extends Transformer {
```

Comment:
This should also be deprecated.

Reply:
I don't think so. There are valid use cases for this, like web apps (they're single threaded), server applications (where you can't spawn isolates for every request).

Renaming and deprecating the old name made a lot of sense though.